### PR TITLE
[Feature] [config] Fix dependency conflict in seatunnel config when running SeatunnelFlink in idea with local mode#1186  

### PR DESCRIPTION
**Pull request by Yves-yuan**
## Purpose of this pull request

This pull request fix the #1186 bug.It add a seatunnel-config-shade module and rename package name of code in seatunnel-config to avoid code conflict when running SeatunnelFlink in Intellij Idea with local mode.

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Yves-yuan** [comment]
@garyelephant Please review.

File: seatunnel-config-shade/pom.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (asdf2014) [inline seatunnel-config-shade/pom.xml]: Please add a license header here

**Review by asdf2014 — CHANGES_REQUESTED** [review]

**Comment by xleoken** [comment]
hi @Yves-yuan, 

if you want to test some case, you can use `LocalFlinkExample`. it seems not necessary to introduce a new config module.

`seatunnel-example` module will run successfully in local env. 

image

**Comment by Yves-yuan** [comment]
> 
Hello,leo65536,
I have tested LocalFlinkExample just now ,it won't throw the error because the order of jar com.typesafe.config is in the front of seatunnel-config\target\classes in classpath.
When I run SeatunnelFlink directly,the order of jar com.typesafe.config is after seatunnel-config\target\classes,so it will throw the error.
So I think it's more reliable to fix this problem.

**Comment by Yves-yuan** [comment]
@CalvinKirs  Hello,code conflict is resolved,can you help rerunning workflow.

**Comment by CalvinKirs** [comment]
> @CalvinKirs Hello,code conflict is resolved,can you help rerunning workflow.

Looks has some errors need resolve, you can look at the log.

**Comment by Yves-yuan** [comment]
> @CalvinKirs Hello,code conflict is resolved,can you help rerunning workflow.

> Looks has some errors need resolve, you can look at the log.
image

The log shows that `Error creating assembly archive bin: /home/runner/work/incubator-seatunnel/incubator-seatunnel/seatunnel-dist/../seatunnel-config/target/seatunnel-config-2.0.5-SNAPSHOT.jar`
`To see the full stack trace of the errors, re-run Maven with the -e switch.`

I have tested mvn -T 2C -B clean verify -Dmaven.test.skip=false -Dgpg.skip=true in my local env,it works well.
Can you rerun maven with -e switch ? @CalvinKirs

**Comment by Yves-yuan** [comment]
image
Hello leo65535,
It seems that the workflow was canceled after 30 minutes.I see that you opened the similar issue in https://github.com/apache/incubator-seatunnel/issues/1202#issue-1126677161.So how can i solve this problem? @leo65535

**Comment by CalvinKirs** [comment]
> > @CalvinKirs Hello,code conflict is resolved,can you help rerunning workflow.
> 
> > Looks has some errors need resolve, you can look at the log.
> > image
> 
> The log shows that `Error creating assembly archive bin: /home/runner/work/incubator-seatunnel/incubator-seatunnel/seatunnel-dist/../seatunnel-config/target/seatunnel-config-2.0.5-SNAPSHOT.jar` `To see the full stack trace of the errors, re-run Maven with the -e switch.`
> 
> I have tested mvn -T 2C -B clean verify -Dmaven.test.skip=false -Dgpg.skip=true in my local env,it works well. Can you rerun maven with -e switch ? @CalvinKirs

rerun~

**Comment by xleoken** [comment]
you can try following statements to trigger the ci again.
```
git commit --allow-empty -m 'trigger rebuild' 
git push
```

**Comment by Yves-yuan** [comment]
> you can try following statements to trigger the ci again.
> 
> ```
> git commit --allow-empty -m 'trigger rebuild' 
> git push
> ```

I have ran these commands,but the workflow didn't start running. 
@leo65535

**Comment by xleoken** [comment]
hi @Yves-yuan, can you check this page https://github.com/Yves-yuan/incubator-seatunnel/actions

**Comment by Yves-yuan** [comment]
> hi @Yves-yuan, can you check this page https://github.com/Yves-yuan/incubator-seatunnel/actions

image
Ok,I checked this page just now.

**Comment by Yves-yuan** [comment]
@asdf2014 
Hi asdf2014,
the checks is passed,can you help reviewing.

**Comment by Yves-yuan** [comment]
@CalvinKirs 
Hi CalvinKirs,
the checks of workflow is canceled cause to timeout,can you help rerunning it?

**Review by xleoken — DISMISSED** [review]
Thanks @Yves-yuan, 💯

**Review by CalvinKirs — CHANGES_REQUESTED** [review]
You need to refer to this PR to modify the corresponding license, because there are changes to the package name.

**Comment by CalvinKirs** [comment]
> You need to refer to this PR to modify the corresponding license, because there are changes to the package name.

https://github.com/apache/incubator-seatunnel/pull/1177/files

**Comment by Yves-yuan** [comment]
> > You need to refer to this PR to modify the corresponding license, because there are changes to the package name.
> 
> https://github.com/apache/incubator-seatunnel/pull/1177/files

@CalvinKirs 
Hi CalvinKirs,
I have changed the path of seatunnel-conf classes in LICENSE file.

**Review by xleoken — DISMISSED** [review]

File: LICENSE [hunk 2]
```diff
@@ -209,8 +209,8 @@ The text of each license is the standard Apache 2.0 license.
...
-seatunnel-config/src/main/java/com/typesafe/config/impl/Path.java              from  https://github.com/lightbend/config
-seatunnel-config/src/main/java/com/typesafe/config/impl/PathParser.java        from  https://github.com/lightbend/config
-seatunnel-config/src/main/java/com/typesafe/config/ConfigParseOptions.java     from https://github.com/lightbend/config
\ No newline at end of file
+seatunnel-config/seatunnel-config-shade/src/main/java/com/typesafe/config/impl/ConfigNodePath.java    from  https://github.com/lightbend/config
+seatunnel-config/seatunnel-config-shade/src/main/java/com/typesafe/config/impl/ConfigParser.java      from  https://github.com/lightbend/config
+seatunnel-config/seatunnel-config-shade/src/main/java/com/typesafe/config/impl/Path.java              from  https://github.com/lightbend/config
+seatunnel-config/seatunnel-config-shade/src/main/java/com/typesafe/config/impl/PathParser.java        from  https://github.com/lightbend/config
+seatunnel-config/seatunnel-config-shade/src/main/java/com/typesafe/config/ConfigParseOptions.java     from https://github.com/lightbend/config
```
Comment (CalvinKirs) [inline LICENSE]: please check file path

**Review by CalvinKirs — APPROVED** [review]
Well done
Looking forward to your next contribution

**Comment by Yves-yuan** [comment]
> Well done Looking forward to your next contribution

Thx,be willing to.

**Comment by xleoken** [comment]
Thanks @CalvinKirs

**Pull request merged**